nilearn/plotting/tests/test_img_plotting.py

```diff
@@ -31,12 +31,39 @@
                        [0.,    0.,    0.,    1.]])
 
 
-def _generate_img():
+@pytest.fixture(autouse=True)
+def _generate_img_3d():
+    """A random 3D image for testing figures.
+    """
     data_positive = np.zeros((7, 7, 3))
     rng = np.random.RandomState(42)
     data_rng = rng.rand(7, 7, 3)
     data_positive[1:-1, 2:-1, 1:] = data_rng[1:-1, 2:-1, 1:]
-    return nibabel.Nifti1Image(data_positive, mni_affine)
+    img_3d = nibabel.Nifti1Image(data_positive, mni_affine)
+    pytest.img_3d = img_3d
+
+
+@pytest.fixture(autouse=True)
+def _generate_img_4d():
+    """A random 4D image for testing figures for multivolume data.
+    """
+    rng = np.random.RandomState(42)
+    data = rng.rand(7, 7, 3, 10)
+    img_4d = nibabel.Nifti1Image(data, mni_affine)
+    pytest.img_4d = img_4d
+
+
+@pytest.fixture(autouse=True)
```

Comment:
I must confess that I find the autouse fixture very implicit and hence hard to read for people who do not know the feature. How easy would it be to use explicit fixtures, rather?

Reply:
I've dropped autouse and just pass the fixtures as arguments to the tests.